samples/nrf9160/modem_shell/src/cloud/cloud_shell.c

```diff
@@ -183,8 +183,8 @@ static void nrf_cloud_event_handler(const struct nrf_cloud_evt *evt)
 #endif
 			/* data was valid; no need to pass to other handlers */
 			break;
-		} else if (err == -EFAULT) {
-			/* data was an A-GPS error; no need to pass to other handlers */
+		} else if ((err == -EFAULT) || (err == -ENOMSG)) {
+			/* not A-GPS data; no need to pass to other handlers */
```

Comment:
Yes, you are right. I am starting to think it would be better to check if the data received is some cloud message not related to A/P-GPS and then break rather than the other way around. I am just not sure how to identify these messages. Should I look for certain attributes such as "pairing"? I wish there was a consistent way to identify every message we receive from the cloud.

Reply:
Indeed, I should have tested this with P-GPS. Moved the check to a different place.

Reply:
unfortunately at this time, all data received from the cloud comes in on one topic under the `NRF_CLOUD_EVT_RX_DATA` event.  until this is changed (i dont recall the Jira ticket), handling the received data from the various cloud components/services is rather ugly.
the data you are seeing, containing "pairing", is part of the cloud connection process.  the P-GPS data is JSON, but is just an array of strings.  the A-GPS and P-GPS error messages use the nrf cloud device message JSON format.
😓

Reply:
Isn't the ticket mentioned by @hiltunent just about this, i.e. having separate topics for different types of data? I don't know if there are already plans when this would be implemented.

Reply:
yes you are correct.  my mistake, i typed in the wrong ticket number.

Reply:
I tried to address all concerns, please check again. I also changed some error codes as @jayteemo proposed. Thanks!